scripts/generator/graphql_generator/csharp/SDK/iOS/iOSWebCheckout.cs.erb

```diff
@@ -0,0 +1,43 @@
+namespace <%= namespace %>.SDK.iOS {
+#if !SHOPIFY_MONO_UNIT_TEST
+    using UnityEngine;
+    using System.Runtime.InteropServices;
+    using <%= namespace %>.SDK;
+
+    class iOSWebCheckout: IWebCheckout {
+        private static iOSWebCheckoutBehaviour webComponent;
+
+        [DllImport ("__Internal")]
+        private static extern void _StartWebCheckout(string unityDelegateObjectName, string checkoutURL);
+
+        public void Start(string checkoutURL, CheckoutSuccessCallback success, CheckoutFailureCallback failure, CheckoutCancelCallback cancelled) {
+            if (webComponent == null) {
+                webComponent = GlobalGameObject.AddComponent<iOSWebCheckoutBehaviour>();
+            }
+        
+            webComponent.OnSuccess = success;
+            webComponent.OnFailure = failure;
+            webComponent.OnCancelled = cancelled;
+
+            _StartWebCheckout(GlobalGameObject.GameObjectName, checkoutURL);
+        }
+    }
+
+    class iOSWebCheckoutBehaviour: MonoBehaviour {
+        internal CheckoutSuccessCallback OnSuccess;
+        internal CheckoutFailureCallback OnFailure;
+        internal CheckoutCancelCallback OnCancelled;
```

Comment:
Since we're creating this behaviour through `AddComponent` we won't be able to pass the values into the constructor. I can probably wrap this in a small private helper method though to clean it up.